src/libraries/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceBase.cs

```diff
@@ -72,6 +72,31 @@ public unsafe void RequestAdditionalTime(int milliseconds)
             }
         }
 
+        /// <summary>
+        /// When this method is called from OnStart, OnStop, OnPause or OnContinue,
+        /// the specified wait hint is passed to the
+        /// Service Control Manager to avoid having the service marked as not responding.
+        /// </summary>
+        /// <param name="time"></param>
+        public unsafe void RequestAdditionalTime(TimeSpan time)
```

Comment:
Please don't duplicate code. One of these overloads can just call into the other one.